src/mrnet/network/reaction_network.py

```diff
@@ -1494,90 +1633,51 @@ def identify_concerted_rxns_for_specific_intermediate(
         :return: list of reactions
         """
 
+        rxn_from_filer_iter1 = []
+        rxn_from_filer_iter1_nodes = []
+        entry_ind = entry.parameters["ind"]  # type: int
+
         if mols_to_keep is None:
             mols_to_keep = list(range(0, len(RN.entries_list)))
         not_wanted_formula = single_elem_interm_ignore
-        reactions_with_in_out_nodes = []
-        unique_reactions = []
-        node = entry.parameters["ind"]
+
         if (
-            RN.entries_list[node].formula not in not_wanted_formula
-            and RN.graph.nodes[node]["bipartite"] == 0
-            and node not in RN.not_reachable_nodes
-            and node not in RN.unsolvable_PRs
+            entry.formula not in not_wanted_formula
+            and entry.parameters["ind"] in mols_to_keep
```

Comment:
I can do that. I tried to keep identify_concerted_reactions_via_intermediates unchanged in case it would create issues with Daniel's code. Maybe in the next PR we can make more design changes while also thinking about iter2?

Reply:
Yes, sounds good